multiapps-controller-persistence/src/main/java/org/cloudfoundry/multiapps/controller/persistence/services/ProcessLoggerProvider.java

```diff
@@ -1,50 +1,78 @@
 import java.util.stream.Collectors;
 
+import javax.inject.Inject;
 import javax.inject.Named;
+import javax.sql.DataSource;
 
 import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.Layout;
+import org.apache.logging.log4j.core.LogEvent;
 import org.apache.logging.log4j.core.Logger;
 import org.apache.logging.log4j.core.LoggerContext;
-import org.apache.logging.log4j.core.appender.FileAppender;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
 import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.apache.logging.log4j.core.filter.LevelMatchFilter;
 import org.apache.logging.log4j.core.layout.AbstractStringLayout;
 import org.apache.logging.log4j.core.layout.PatternLayout;
-import org.cloudfoundry.multiapps.common.SLException;
 import org.cloudfoundry.multiapps.controller.persistence.Constants;
 import org.cloudfoundry.multiapps.controller.persistence.Messages;
+import org.cloudfoundry.multiapps.controller.persistence.model.ImmutableOperationLogEntry;
+import org.cloudfoundry.multiapps.controller.persistence.model.OperationLogEntry;
+import org.cloudfoundry.multiapps.controller.persistence.query.providers.SqlOperationLogQueryProvider;
 import org.flowable.engine.delegate.DelegateExecution;
 
 @Named("processLoggerProvider")
 public class ProcessLoggerProvider {
 
     static final String LOG_LAYOUT = "#2.0#%d{yyyy MM dd HH:mm:ss.SSS}#%d{XXX}#%p#%c#%n%X{MsgCode}#%X{CSNComponent}#%X{DCComponent}##%X{DSRCorrelationId}#%X{Application}#%C#%X{User}#%X{Session}#%X{Transaction}#%X{DSRRootContextId}#%X{DSRTransaction}#%X{DSRConnection}#%X{DSRCounter}#%t##%X{ResourceBundle}#%n%m#%n%n";
-
+    private static final LevelMatchFilter DEBUG_FILTER = LevelMatchFilter.newBuilder()
+                                                                         .setLevel(Level.DEBUG)
+                                                                         .setOnMatch(Filter.Result.ACCEPT)
+                                                                         .setOnMismatch(Filter.Result.ACCEPT)
+                                                                         .build();
     private static final String PARENT_LOGGER = "com.sap.cloud.lm.sl.xs2";
     private static final String DEFAULT_LOG_NAME = "OPERATION";
-    private static final String DEFAULT_LOG_DIR = "logs";
     private static final String LOG_FILE_EXTENSION = ".log";
+    private final DataSource dataSource;
 
     private final Map<String, ProcessLogger> loggersCache = new ConcurrentHashMap<>();
+    private final LoggerContext loggerContext = (LoggerContext) LogManager.getContext(false);
+
+    @Inject
+    public ProcessLoggerProvider(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+
+    public ProcessLoggerProvider() {
+        this.dataSource = null;
```

Comment:
why is this needed?

Reply:
for unit tests. without this there are about 50 tests that are failing because i cant create mock of processloggerprovider because it doens't have empty contructor